services/kaze/middlewares/instance_authorization.go

```diff
@@ -28,7 +28,7 @@ func VerifyAdmin(c *gin.Context) {
 }
 
 func isInstanceOwner(c *gin.Context, instanceType string) {
-	instance := c.Param(instanceType)
+	instance := c.Param("app")
```

Comment:
It can be either "app" or "db" depending upon whether the middleware is "IsAppOwner" or "IsDatabaseOwner". Include this as a function parameter to "isInstanceOwner" and wrap over it via IsAppOwner and IsDatabaseOwner